quic/transport/ngtcp2/stream/openstate.nim

```diff
@@ -49,9 +49,9 @@ method read(state: OpenStream): Future[seq[byte]] {.async.} =
   state.allowMoreIncomingBytes(result.len.uint64)
 
 method write(state: OpenStream, bytes: seq[byte]): Future[void] =
-  let stream = state.stream.valueOr:
-    raise newException(QuicError, "stream is closed")
-  state.connection.send(stream.id, bytes)
+  # let stream = state.stream.valueOr:
+  #   raise newException(QuicError, "stream is closed")
```

Comment:
I don't get why you removed those two lines. If stream is none, it'll raise an error nevertheless. It'll just be less precise.

Reply:
With this code, there's a compilation error: `Error: (ref QuicError)(msg: "stream is closed", parent: nil) can raise an unlisted exception: ref QuicError`. Would you happen to know how to fix it?

Reply:
```nim
method write(state: OpenStream, bytes: seq[byte]) {.async.} =
  let stream = state.stream.valueOr:
    raise newException(QuicError, "stream is closed")
  return state.connection.send(state.stream.get.id, bytes)
 ```
 fails with:
` Error: undeclared identifier: 'setResult'`

Reply:
I dug a bit into this. Here a solution:
```nim
method write(state: OpenStream, bytes: seq[byte]): Future[void] {.raises: [QuicError].} =    
  let stream = state.stream.valueOr:    
    raise newException(QuicError, "stream is closed")    
  state.connection.send(state.stream.get.id, bytes)    
```
With the base method modified aswell:
```nim
method write*(state: StreamState, bytes: seq[byte]) {.base, async, raises: [QuicError].} =    
  doAssert false # override this method                                     
```
But I'm not satisfied at all about this. It uses an old version of chronos and thus is not up to date regarding exception. With the newer version (> 4.0 I think), you'll use `{.async: (raises: [QuicError]).}`.
It uses aswell `upraises` which I just discover, it seems to be an exception tracking for older versions of nim... Which I'm hesitant to use aswell. 

But it's clearly not the purpose of this PR. So I might just approve if you have nothing else to add.

Reply:
Changing the base method would require changing all the other implementations, I guess. I'm not sure yet if we should raise this error. As you said, not the point of this PR.

Reply:
Added a link to this PR